test/e2e/scheduling/priorities.go

```diff
@@ -254,6 +265,30 @@ var _ = SIGDescribe("SchedulerPriorities [Serial]", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
 	})
+	It("Pod should be preferably scheduled to nodes which satisfy its limits", func() {
+		// Update one node to have large allocatable.
+		nodeName := nodeList.Items[0].Name
+		err := UpdateMemoryOfNode(cs, nodeName)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = createBalancedPodForNodes(f, cs, ns, nodeList.Items, podRequestedResource, 0.5)
+		framework.ExpectNoError(err)
+		By("Create a pod with unusual large limits")
+		podWithLargeLimits := "with-large-limits"
+
+		pod := createPausePod(f, pausePodConfig{
+			Name:      podWithLargeLimits,
+			Resources: podwithLargeRequestedResource,
+		})
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		By("Pod should preferbly scheduled to nodes which satisfy its limits")
+		tolePod, err := cs.CoreV1().Pods(ns).Get(podWithLargeLimits, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(tolePod.Spec.NodeName).To(Equal(nodeName))
```

Comment:
I think you need to revert the node resources back to avoid surprises for future test cases.

Reply:
Yes, I missed it. I have updated the PR now.